### ISSUE TITLE
Ratdata/Rat Issue: ConventionalFighter Rat

### ISSUE THREAD

**Issue by AlvarOldebrecht** [MegaMek/mekhq#1302]
### Environment
MekHQ 45,5 and MekHQ46.0
Windows 10
Java version 1.8.0_221 or Java version 12.0.2

### Description
At the moment i'm creating a "Age of War"-RAT and found an error. If i put in the ratdata xml-file the ConventionalFighter-RATs above Mek-RATs, MekHQ will always roll for Meks on the ConventionalFighter-RAT. 

![ConventionalFigher Rat above Mek](https://user-images.githubusercontent.com/56050031/66003402-53043700-e4a6-11e9-941b-5b35f6c23871.png)

The Error is fixed, when the Mek-RATs are written above the ConventionalFighter-RATs.

![Mek Rat above ConventionalFigher Rat](https://user-images.githubusercontent.com/56050031/66003425-60212600-e4a6-11e9-907f-b89d11093824.png)

But as far as i know, this shouldn't happen.

### Files

[My Campaign25300320.zip](https://github.com/MegaMek/mekhq/files/3678630/My.Campaign25300320.zip)
[mekhqlog.txt](https://github.com/MegaMek/mekhq/files/3678608/mekhqlog.txt)
[MekHQ46.zip](https://github.com/MegaMek/mekhq/files/3678615/MekHQ46.zip)

**Comment by Windchild292** [comment]
I've looked into this, and, while it has resulted in some code cleanup, I've not found the cause of this issue. I will be opening a PR for this code cleanup, however.

I've also found this issue occurs when you try to select a weight class of mech that is not part of the RAT. In the aforementioned case, it appears that it is defaulting to a ConventionalFighter roll or selecting the top ConventionalFighter instead of saying nothing can be found.

Based on my read of the code, I believe this may be an issue in MegaMek, not in MekHQ.
